### PR TITLE
Comment out explicit check against M2_PATTERN

### PR DESCRIPTION
**Pull request by eed3si9n**
REF sbt/sbt#2005, and potentially relates to some of the SNAPSHOT issues listed on sbt/sbt#1780.

Ivy checks if the repository patterns ends with `M2_PATTERN` to see if `maven-metadata.xml` should be used. Because sbt customizes the pattern, Ivy will end up not using `maven-metadata.xml`.

/review @jsuereth, @dwijnand

### CODE REVIEW

File: build.sbt [hunk 1]
```diff
@@ -6,10 +6,23 @@ lazy val makeModuleProperties = taskKey[Seq[File]]("Create module.properties fil
 lazy val root = (project in file(".")).
   settings(versionWithGit: _*).
   settings(
+    inThisBuild(Seq(
```
Comment (dwijnand) [inline build.sbt:9]: hello! :)

**Comment by dwijnand** [comment]
Sorry I've no idea what changing `shouldUseMavenMetadata` would impact.

**Comment by eed3si9n** [comment]
@dwijnand I appreciate the candid honesty. It's impossible to keep track of all the details, so feel free to ask for explanation of what's going on. Note that these are facts that I've dug out recently, so might not be 100% accurate.

When Ivy sees dynamic revisions like version range, it queries the repository for the list of all available versions for the given artifact. The method that gets called eventually is `listResources` - https://github.com/sbt/ivy/blob/aa94ce736984ceb123a327ae5e7a3269baadbf05/src/java/org/apache/ivy/plugins/resolver/IBiblioResolver.java#L392-L433

It first determines that the repository is fully compatible with Maven using `shouldUseMavenMetadata(pattern)`, if it is, it uses `maven-metadata.xml` that lists out the versions. Otherwise, it tries screen scraping using [ApacheURLLister](https://github.com/sbt/ivy/blob/aa94ce736984ceb123a327ae5e7a3269baadbf05/src/java/org/apache/ivy/util/url/ApacheURLLister.java). The Apache screen scraping doesn't work with Bintray, because they've added "#" character to their directory listing. The goal here to make `shouldUseMavenMetadata(pattern)` more lenient so sbt's dependency resolution can use more accurate `maven-metadata.xml`.

**Comment by dwijnand** [comment]
But why does sbt (not ivy) change the maven pattern?

Also, given a release off of the head of this PR is now in sbt master, this should be merged.

**Comment by eed3si9n** [comment]
sbt's pattern encodes `(_[scalaVersion])(_[sbtVersion])` so we can do cross publishing in Maven sort-of-compatible way. - https://github.com/sbt/sbt/blob/0.13/ivy/src/main/scala/sbt/Resolver.scala#L328-L330

**Comment by dwijnand** [comment]
Oh, those. I see. Huh, but that was years ago! Ok thanks.

**Pull request merged**